r-package/grf/R/local_linear_forest.R

```diff
@@ -68,48 +79,124 @@
-                              num.threads = num.threads,
-                              seed = seed)
-  forest[["tuning.output"]] <- NULL
+                                ll.splits = FALSE,
+                                ll.split.weight.penalty = FALSE,
+                                ll.split.lambda = 0.1,
+                                ll.split.variables = NULL,
+                                ll.split.cutoff = NULL,
+                                num.trees = 2000,
+                                sample.weights = NULL,
+                                clusters = NULL,
+                                equalize.cluster.weights = FALSE,
+                                sample.fraction = 0.5,
+                                mtry = min(ceiling(sqrt(ncol(X)) + 20), ncol(X)),
+                                min.node.size = 5,
+                                honesty = TRUE,
+                                honesty.fraction = 0.5,
+                                honesty.prune.leaves = TRUE,
+                                alpha = 0.05,
+                                imbalance.penalty = 0,
+                                ci.group.size = 2,
+                                tune.parameters = "none",
+                                tune.num.trees = 50,
+                                tune.num.reps = 100,
+                                tune.num.draws = 1000,
+                                num.threads = NULL,
+                                seed = runif(1, 0, .Machine$integer.max)) {
+
+  validate_X(X)
+  validate_sample_weights(sample.weights, X)
+  Y <- validate_observations(Y, X)
+  clusters <- validate_clusters(clusters, X)
+  samples.per.cluster <- validate_equalize_cluster_weights(equalize.cluster.weights, clusters, sample.weights)
+  num.threads <- validate_num_threads(num.threads)
+
+  ll.split.variables <- validate_ll_vars(ll.split.variables, ncol(X))
+  ll.split.lambda <- validate_ll_lambda(ll.split.lambda)
+  ll.split.cutoff <- validate_ll_cutoff(ll.split.cutoff, nrow(X))
+
+  all.tunable.params <- c("sample.fraction", "mtry", "min.node.size", "honesty.fraction",
+                          "honesty.prune.leaves", "alpha", "imbalance.penalty")
+
+  data <- create_data_matrices(X, outcome = Y, sample.weights = sample.weights)
+
+  args <- list(num.trees = num.trees,
+               clusters = clusters,
+               samples.per.cluster = samples.per.cluster,
+               sample.fraction = sample.fraction,
+               mtry = mtry,
+               min.node.size = min.node.size,
+               honesty = honesty,
+               honesty.fraction = honesty.fraction,
+               honesty.prune.leaves = honesty.prune.leaves,
+               alpha = alpha,
+               imbalance.penalty = imbalance.penalty,
+               ci.group.size = ci.group.size,
+               num.threads = num.threads,
+               seed = seed)
+  if (ll.splits & ll.split.cutoff > 0) {
```

Comment:
use && to be consistent with rest of grf? (see for example input_utilities.R, and update || in validate_ll_cutoff too?)